### PR TITLE
Sowmya| Phani | BAH-3113 | Adds React components for the right panel of the Drug Chart (MAR)

### PR DESCRIPTION
**Pull request by sowmya-AS**
Link to JIRA card  -
https://bahmni.atlassian.net/jira/software/c/projects/BAH/boards/55?modal=detail&selectedIssue=BAH-3113

### CODE REVIEW

**Review by Arjun-Go — APPROVED** [review]

File: src/features/Calendar/Calendar.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import PropTypes from "prop-types";
+import CalendarRow from "../CalendarRow/CalendarRow.jsx";
+import "./Calendar.scss";
+export default function Calendar(props) {
+  const { calendarData } = props;
+  return (
+    <table className={"drug-chart-calendar"}>
```
Comment (bassoGeorge) [inline src/features/Calendar/Calendar.jsx:8]: If it is a string, you don't need this unnecessary variable wrapping. you can simply do className="string"

File: src/features/DrugChart/DrugChart.jsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
+import React from "react";
+import PropTypes from "prop-types";
+import moment from "moment";
+import Calendar from "../Calendar/Calendar";
+import CalendarHeader from "../CalendarHeader/CalendarHeader";
+import "./DrugChart.scss";
+
+export const TransformDrugChartData = (drugChartData) => {
+  const transformedDrugChartData = [];
+  drugChartData.forEach((schedule) => {
```
Comment (bassoGeorge) [inline src/features/DrugChart/DrugChart.jsx:10]: Coding nitpick. using a forEach to run a process on each item in an array to push a transformed into into another array is the EXACT usecase of a .map() function. Please use map, that is more clear

File: src/features/DrugChart/DrugChart.jsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    const { slots } = schedule;
+    const slotData = {};
+    slots.forEach((slot) => {
```
Comment (bassoGeorge) [inline src/features/DrugChart/DrugChart.jsx:13]: Another coding nitpick, although this one is less of an issue. Here you are processing an array into an object. that is the use case for a .reduce() function. If you are comfortable with reduce, you can use that else let this one be.

**Review by bassoGeorge — APPROVED** [review]
Take a call whether you want to implement the changes I requested

**Pull request merged**